### PR TITLE
Refactor to make wrapper functions for most generic queries, add missing APIs

### PR DESCRIPTION
**Pull request by jcbang**
Refactoring:
- QueryStrings.cs consists of wrapper functions to abstract away most queries. API Controllers now call QueryStrings functions instead of constructing queries per-controller.
- QueryBuilder.cs consists of individual query commands utilized in QueryString.cs.
- Utility.cs now consists of generic helper functions not necessarily tied to querying.

Missing APIs:
- Added generic "Like / Unlike" API
- Added querying for all samples, beats, or playlists
- Added generic "Get Owned / Get Liked" API

### CODE REVIEW

**Review by gabeeee — DISMISSED** [review]
Looks good

File: brainbeats-backend/Controllers/BeatController.cs [hunk 1]
```diff
@@ -13,189 +14,124 @@ public class BeatController : ControllerBase
...
+      } catch {
+        return BadRequest("Malformed Request");
       }
 
       try {
         var result = await DatabaseConnection.Instance.ExecuteQuery(queryString.ToString());
         return Ok(result);
       } catch {
-        return BadRequest();
+        return BadRequest("Something went wrong");
```
Comment (WarmBuns) [inline brainbeats-backend/Controllers/BeatController.cs:35]: Is it possible to give a more helpful error message?

File: brainbeats-backend/Controllers/BeatController.cs [hunk 2]
```diff
@@ -13,189 +14,124 @@ public class BeatController : ControllerBase
...
 
-      string beatId = body.beatId;
+      string queryString;
 
-      if (beatId == null) {
-        return BadRequest("Malformed Request");
+      try {
+        queryString = ReadVertexQuery(body.GetValue("beatId").ToString());
+      } catch {
+        return BadRequest("Malformed request");
```
Comment (WarmBuns) [inline brainbeats-backend/Controllers/BeatController.cs:49]: Can we say what about the request was Malformed? Maybe "Invalid beatId"

File: brainbeats-backend/Controllers/BeatController.cs [hunk 3]
```diff
@@ -13,189 +14,124 @@ public class BeatController : ControllerBase
...
+      JObject body = DeserializeRequest(req);
+
+      string queryStringPublic;
+      string queryStringPrivate;
+
+      try {
+        queryStringPublic = GetAllPublicVerticesQuery("beat");
+        queryStringPrivate = GetAllPrivateVerticesQuery("beat", body.GetValue("email").ToString());
+      } catch {
+        return BadRequest("Malformed request");
```
Comment (WarmBuns) [inline brainbeats-backend/Controllers/BeatController.cs:72]: same as above

File: brainbeats-backend/Controllers/BeatController.cs [hunk 4]
```diff
@@ -13,189 +14,124 @@ public class BeatController : ControllerBase
...
+          resultList.Add(item);
+        }
 
-      try {
-        var result = await DatabaseConnection.Instance.ExecuteQuery(queryString);
-        return Ok(result);
+        return Ok(resultList);
       } catch {
-        return BadRequest();
+        return BadRequest("Something went wrong");
```
Comment (WarmBuns) [inline brainbeats-backend/Controllers/BeatController.cs:91]: Same as previous comments

File: brainbeats-backend/Controllers/TestController.cs [hunk 5]
```diff
@@ -16,14 +15,13 @@ public class TestController : ControllerBase
     [HttpPost]
     [Route("create")]
     public async Task<IActionResult> CreateSeed(dynamic req) {
```
Comment (WarmBuns) [inline brainbeats-backend/Controllers/TestController.cs]: What exactly is this test controller testing?

File: brainbeats-backend/DatabaseConnection.cs [hunk 6]
```diff
@@ -33,7 +33,7 @@ public class DatabaseConnection {
 
     public Task<ResultSet<dynamic>> ExecuteQuery(string query) {
       try {
-          return gremlinClient.SubmitAsync<dynamic>(query);
+         return gremlinClient.SubmitAsync<dynamic>(query);
       } catch (ResponseException e) {
         Console.WriteLine("\tRequest Error!");
```
Comment (WarmBuns) [inline brainbeats-backend/DatabaseConnection.cs:38]: Could we have a flag that in DEBUG mode it sends the exception and in production we have generic errors? We can follow that pattern with all exceptions honestly

**Review by WarmBuns — CHANGES_REQUESTED** [review]

File: brainbeats-backend/Controllers/TestController.cs (hunk 5 above)
Comment (jcbang) [reply brainbeats-backend/Controllers/TestController.cs]: Fixed: TestController is renamed to SeedController to better represent its purpose: seeding data into the database

File: brainbeats-backend/Controllers/BeatController.cs (hunk 1 above)
Comment (jcbang) [reply brainbeats-backend/Controllers/BeatController.cs:35]: Will create a new issue to enable Azure Application Insights -> This is too large of a change to include in this PR

**Review by WarmBuns — DISMISSED** [review]

**Pull request merged**